### PR TITLE
Change schedule and WS conn print behavior

### PR DESCRIPTION
**Pull request by MilesHolland**
# Description

Changing the `__str__` method of the resource class (base class of all entity classes). The change adds a new kwarg which, if true, makes the entity's `print(entity_instance)` behavior produce a string that's identical to what's shown by most entity's CLI `show` command. By default, this kwarg is only set to true when running within a python notebook, and remains false (and thus continues to use the existing print behavior) for all other cases, including our tests.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Features Added
 - Added OS Patching Parameters to Compute Instance.
+- Change print behavior of various entity classes to match CLI `show` behavior.
```
Comment (harneetvirk) [inline sdk/ml/azure-ai-ml/CHANGELOG.md:7]: Please add the entry for this change for the upcoming SDK release. SDK 1.2.0 cannot have this change.

**Comment by harneetvirk** [comment]
To unblock the validation gate, you may need to update the _version.py file to update the version as we have in the CHANGELOG file.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py [hunk 2]
```diff
@@ -184,4 +193,8 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({var_dict})"
 
     def __str__(self) -> str:
-        return self.__repr__()
+        if self.print_as_yaml:
+            yaml_serialized = self._to_dict()
+            return dump_yaml(yaml_serialized, default_flow_style=False)
+        else:
+            return self.__repr__()
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py:200]: It's prob never a good idea to show __repr__ for __str__. E,g. class name shouldn't be part of __str__.

Once we implement DictMixin for all entity classes. I'm wondering if we should just print the dictionary for __str__.

File: sdk/ml/azure-ai-ml/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 1.2.1 (Unreleased)
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/CHANGELOG.md:3]: Next release version should be 1.3.0?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py (hunk 2 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py:200]: That's true, but I can't remove the use of repr easily for two reasons:
 - It was the existing implementation we already had.
 - A bunch of tests have been built to rely on the output it produces.

File: sdk/ml/azure-ai-ml/CHANGELOG.md (hunk 3 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/CHANGELOG.md:3]: Oh it should, I'll change that then. I thought we were doing minor versions.

Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/CHANGELOG.md:3]: Looking at our changelog, I can see that we've done tertiary versions before. I'm not sure when we choose to do tertiary versus secondary version updates. I'll consult @harneetvirk  about this.

Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/CHANGELOG.md:3]: According to Harneet, this should be a 1.3 version, since we aren't doing a patch release.

**Review by luigiw — APPROVED** [review]

**Review by nthandeMS — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py [hunk 4]
```diff
@@ -29,6 +32,7 @@ def __init__(
         description: Optional[str] = None,
         tags: Optional[Dict] = None,
         properties: Optional[Dict] = None,
+        print_as_yaml: Optional[bool] = None,
```
[the remark this answers is not in the document]
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py]: Removed the function header reference, but kept and fixed the docstring description.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py [hunk 5]
```diff
@@ -41,13 +45,18 @@ def __init__(
         :type tags: Dict, optional
         :param properties: The asset property dictionary., defaults to None
         :type properties: Dict, optional
+        param: print_as_yaml: If set to true, then printing out this resource will produce a YAML-formatted object.
+            False will force a more-compact printing style. By default, the YAML output is only used in jupyter
+            notebooks. Be aware that some bookkeeping values are shown only in the non-YAML output.
+        type print_as_yaml: bool, optional
```
[the remark this answers is not in the document]
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py]: Fixed

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py [hunk 6]
```diff
@@ -41,13 +45,18 @@ def __init__(
         :type tags: Dict, optional
         :param properties: The asset property dictionary., defaults to None
         :type properties: Dict, optional
+        param: print_as_yaml: If set to true, then printing out this resource will produce a YAML-formatted object.
```
[the remark this answers is not in the document]
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py]: Fixed

**Pull request merged**